### ISSUE TITLE
Auto-collect schema references for IntoResponses-style responses

### ISSUE THREAD

**Issue by plv** [juhaku/utoipa#1224]
Hi, thanks for all the hard work on `utoipa`!

In #1071 I see you added auto collection of tuple responses for `utoipa::path` if the responses are tuple-style (i.e. `responses((status = 200, description = "200"))`, but that `IntoResponses`-style (i.e. `responses(FooStruct)` is [not supported](https://github.com/juhaku/utoipa/blob/fa48b879085b1984d792a28701e922ecaf385915/utoipa-gen/src/path/response.rs#L79). What happens is that if the response is a struct with named fields has nested structs (with their own respective `ToSchema` implementations, there will be hanging `$ref`s that are not included. My current workaround is to force inlining with `#[schema(inline)]` on these fields, but I'd much prefer if the `$refs` were automatically collected

Is this a misuse/misunderstanding of `IntoResponses`? Or will rescursive and automatic components/`$ref` collection be implemented for `IntoResponses` in the future?

Thanks again!

**Comment by juhaku** [comment]
True, this is currently still missing from the implementation and there is still probably big rework ahead of before supporting this for `IntoResponses` and `ToResponse` derive macros. But the plan is to have it supported as well eventually.

**Comment by immno** [comment]
Can it be implemented by referring to https://github.com/salvo-rs/salvo/tree/main/crates/oapi?
